eCommerce platforms/Adobe Magento/src/code/local/Microsoft/Dfp/Helper/Data.php

```diff
@@ -0,0 +1,161 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Microsoft
+ * @package    Dynamics 365 Fraud Protection
+ * @copyright  Copyright (c) Microsoft Corporation. (https://www.microsoft.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+class Microsoft_Dfp_Helper_Data extends Mage_Core_Helper_Abstract
+{
+	public function GUID()
+	{
+		if (function_exists('com_create_guid')) {
+			return trim(com_create_guid(), '{}');
+		}
+		return sprintf('%04X%04X-%04X-%04X-%04X-%04X%04X%04X', mt_rand(0, 65535), mt_rand(0, 65535), mt_rand(0, 65535), mt_rand(16384, 20479), mt_rand(32768, 49151), mt_rand(0, 65535), mt_rand(0, 65535), mt_rand(0, 65535));
+	}
+
+	public function log($msg)
+	{
+		if (Mage::getStoreConfigFlag('dfp/general/enablelog')) {
+			Mage::log($msg, null, 'MicrosoftDFP.log');
+		}
+	}
+
+	public function getFingerprintingSessionId()
+	{
+		$sessionId = Mage::getSingleton('core/session')->getFptDfpSessionId();
+		if (empty($sessionId)) {
+			$sessionId = $this->GUID();
+			Mage::getSingleton('core/session')->setFptDfpSessionId($sessionId);
+		}
+		return $sessionId;
+	}
+
+	public function sendUpdateAccount($payload, $correlationId = null)
+	{
+		return $this->invokeDFP('dfp/endpoints/updateaccount', $payload, $correlationId);
+	}
+
+	public function sendPurchase($payload, $correlationId = null)
+	{
+		return $this->invokeDFP('dfp/endpoints/purchase', $payload, $correlationId);
+	}
+
+	public function sendBankEvent($payload, $correlationId = null)
+	{
+		return $this->invokeDFP('dfp/endpoints/bankevent', $payload, $correlationId);
+	}
+
+	public function sendPurchaseStatus($payload, $correlationId = null)
+	{
+		return $this->invokeDFP('dfp/endpoints/purchasestatus', $payload, $correlationId);
+	}
+
+	public function sendRefund($payload, $correlationId = null)
+	{
+		return $this->invokeDFP('dfp/endpoints/refund', $payload, $correlationId);
+	}
+
+	public function sendChargeback($payload, $correlationId = null)
+	{
+		return $this->invokeDFP('dfp/endpoints/chargeback', $payload, $correlationId);
+	}
+
+	private function invokeDFP($configName, $payload, $correlationId)
+	{
+		$apiMethod = Mage::getStoreConfig($configName);
+		$correlationId = $correlationId ?: $this->GUID();
+		$jsonPayload = json_encode($payload);
+
+		$this->log("Payload " . $apiMethod . ": " . $jsonPayload);
+		$this->log("Correlation Id: " . $correlationId);
+
+		$accessToken = $this->getAccessToken();
+		if (empty($accessToken)) {
+			$this->log("Failed to get access token!");
+			throw new Exception("Failed to get access token!");
+		}
+
+		$this->log("Access Token: " . $accessToken);
+
+		$headers = array();
+		$headers[] = 'x-ms-correlation-id: ' . $correlationId;
```

Comment:
Is php string interpolation still as bad as it used to be?

Reply:
Here's a bit of reading if you want to know.  I think for this small amount of concatenation, the difference isn't so important as consistency within the code base is.  

https://stackoverflow.com/questions/16790501/php-variable-interpolation-vs-concatenation
https://stackoverflow.com/questions/4676417/should-i-use-curly-brackets-or-concatenate-variables-within-strings